cherab/core/laser/models/laserspectrum_base.pyx

```diff
@@ -0,0 +1,125 @@
+
+from raysect.core.math.function cimport Function1D
+from raysect.optical cimport Point3D, Vector3D
+
+from cherab.core.utility import Notifier
+from cherab.core.laser.node import Laser
+import numpy as np
+cimport numpy as np
+
+from cherab.core.utility.constants cimport SPEED_OF_LIGHT, PLANCK_CONSTANT
+cdef double CONSTANT_HC = SPEED_OF_LIGHT * PLANCK_CONSTANT
+
+cdef class LaserSpectrum(Function1D):
+
+    def __init__(self, double min_wavelength, double max_wavelength, int bins):
+
+        super().__init__()
+
+        self._notifier = Notifier()
+
+        self._check_wavelength_validity(min_wavelength, max_wavelength)
+
+        self._min_wavelength = min_wavelength
+        self._max_wavelength = max_wavelength
+
+        self.bins = bins
+
+
+    @property
+    def min_wavelength(self):
+        return self._min_wavelength
+
+    @min_wavelength.setter
+    def min_wavelength(self, double value):
+
+        self._min_wavelength = value
+        self._update_cache()
+
+    @property
+    def max_wavelength(self):
+        return self._max_wavelength
+
+    @max_wavelength.setter
+    def max_wavelength(self, double value):
+
+        self._max_wavelength = value
+        self._update_cache()
+
+    @property
+    def bins(self):
+        return self._bins
+
+    @bins.setter
+    def bins(self, int value):
+        if not value > 0:
+            raise ValueError("Value has to be larger than 0")
+
+        self._bins = value
+        self._update_cache()
+
+    @property
+    def wavelengths(self):
+        return self._wavelengths
+
+    @property
+    def power_spectral_density(self):
+        return self._power_spectral_density
+
+    @property
+    def photons(self):
+        return self._photons
+
+    @property
+    def delta_wavelength(self):
+        return self._delta_wavelength
+
+    def _check_wavelength_validity(self, min_wavelength = None, max_wavelength = None):
```

Comment:
PEP8 suggests no space between argument, equals sign and value in function definitions and calls (unless type hints are provided, which isn't the case here). This should be e.g. `min_wavelength=None`.